Source/WebCore/dom/Position.cpp

```diff
@@ -696,40 +704,41 @@ Position Position::upstream(EditingBoundaryCrossingRule rule) const
         return { };
     
     // iterate backward from there, looking for a qualified position
-    Node* boundary = enclosingVisualBoundary(startNode.get());
+    RefPtr boundary = enclosingVisualBoundary(startNode.get());
     // FIXME: PositionIterator should respect Before and After positions.
-    PositionIterator lastVisible = m_anchorType == PositionIsAfterAnchor ? makeDeprecatedLegacyPosition(m_anchorNode.get(), caretMaxOffset(*m_anchorNode)) : *this;
+    auto anchorNode = protectedAnchorNode();
```

Comment:
Personally, I was using `RefPtr anchorNode = m_anchorNode;` or `RefPtr anchorNode = anchorNode();`. This is fine too but I felt the `protectedX()` functions were mostly useful to avoid local variables and pass the result directly as function parameter.

Reply:
Okay, we can do: `RefPtr anchorNode = m_anchorNode;`